scripts/preinstall.js

```
@@ -1,78 +1,90 @@
       }
       console.log(message);
     });
-});
+  });
 }
 
-if (process.platform === 'win32') {
-  var LIB_URL = 'https://github.com/nteract/libzmq-win/releases/download/v2.0.0/libzmq-' + ZMQ + '-' + process.arch + '.lib';
-  var DIR_NAME = path.join(__dirname, '..', 'windows', 'lib');
-  var FILE_NAME = path.join(DIR_NAME, 'libzmq.lib');
+if (process.platform === "win32") {
+  var LIB_URL =
+    "https://github.com/nteract/libzmq-win/releases/download/v2.0.0/libzmq-" +
+    ZMQ +
+    "-" +
+    process.arch +
+    ".lib";
+  var DIR_NAME = path.join(__dirname, "..", "windows", "lib");
+  var FILE_NAME = path.join(DIR_NAME, "libzmq.lib");
 
   if (!fs.existsSync(DIR_NAME)) {
     fs.mkdirSync(DIR_NAME);
   }
 
   if (!fs.existsSync(FILE_NAME)) {
-    console.log('Downloading libzmq for Windows');
+    console.log("Downloading libzmq for Windows");
     download(LIB_URL, FILE_NAME, function() {
-      console.log('Download finished');
+      console.log("Download finished");
     });
   }
-
 } else {
-  var SCRIPT_PATH = path.join(__dirname, 'build_libzmq.sh');
-  var TAR_URL = 'https://github.com/zeromq/' + ZMQ_REPO + '/releases/download/v' + ZMQ + '/zeromq-' + ZMQ + '.tar.gz';
-  var DIR_NAME = path.join(__dirname, '..', 'zmq');
-  var FILE_NAME = path.join(DIR_NAME, 'zeromq-' + ZMQ + '.tar.gz');
+  var SCRIPT_PATH = path.join(__dirname, "build_libzmq.sh");
+  var TAR_URL =
+    "https://github.com/zeromq/" +
+    ZMQ_REPO +
+    "/releases/download/v" +
+    ZMQ +
+    "/zeromq-" +
+    ZMQ +
+    ".tar.gz";
+  var DIR_NAME = path.join(__dirname, "..", "zmq");
+  var FILE_NAME = path.join(DIR_NAME, "zeromq-" + ZMQ + ".tar.gz");
 
   if (!fs.existsSync(DIR_NAME)) {
     fs.mkdirSync(DIR_NAME);
   }
 
-  if (fs.existsSync(path.join(DIR_NAME, 'BUILD_SUCCESS'))) {
-    return console.log('Libzmq found, skipping rebuild.');
+  if (fs.existsSync(path.join(DIR_NAME, "BUILD_SUCCESS"))) {
+    console.log("Libzmq found, skipping rebuild.");
+    process.exit(0);
```

Wouldn't `process.exit()` hide errors that leave pending async operations?